src/ethereum/prague/requests.py

```diff
@@ -13,28 +13,125 @@
 
 from ethereum_types.bytes import Bytes
 
+from ethereum.exceptions import InvalidBlock
+from ethereum.utils.hexadecimal import hex_to_bytes32
+
 from .blocks import Receipt, decode_receipt
 from .utils.hexadecimal import hex_to_address
 
 DEPOSIT_CONTRACT_ADDRESS = hex_to_address(
     "0x00000000219ab540356cbb839cbe05303d7705fa"
 )
+DEPOSIT_EVENT_SIGNATURE_HASH = hex_to_bytes32(
+    "0x649bbc62d0e31342afea4e5cd82d4049e7e1ee912fc0889aa790803be39038c5"
+)
 DEPOSIT_REQUEST_TYPE = b"\x00"
 WITHDRAWAL_REQUEST_TYPE = b"\x01"
 CONSOLIDATION_REQUEST_TYPE = b"\x02"
 
 
+DEPOSIT_EVENT_LENGTH = 576
+
+PUBKEY_OFFSET = 160
+WITHDRAWAL_CREDENTIALS_OFFSET = 256
+AMOUNT_OFFSET = 320
+SIGNATURE_OFFSET = 384
+INDEX_OFFSET = 512
+
+PUBKEY_SIZE = 48
+WITHDRAWAL_CREDENTIALS_SIZE = 32
+AMOUNT_SIZE = 8
+SIGNATURE_SIZE = 96
+INDEX_SIZE = 8
+
+
 def extract_deposit_data(data: Bytes) -> Bytes:
     """
     Extracts Deposit Request from the DepositContract.DepositEvent data.
+
+    Raises
+    ------
+    InvalidBlock :
+        If the deposit contract did not produce a valid log.
     """
-    return (
-        data[192:240]  # public_key
-        + data[288:320]  # withdrawal_credentials
-        + data[352:360]  # amount
-        + data[416:512]  # signature
-        + data[544:552]  # index
+    if len(data) != DEPOSIT_EVENT_LENGTH:
+        raise InvalidBlock("Invalid deposit event data length")
+
+    # Check that all the offsets are in order
+    pubkey_offset = int.from_bytes(data[0:32], "big")
```

Comment:
Any reason to not use [`U256.from_be_bytes`](https://ethereum.github.io/ethereum-types/src/ethereum_types/numeric.py.html#ethereum_types.numeric.FixedUnsigned.from_be_bytes:0)?

Reply:
Only that the [specs](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-6110.md#block-validity) do explicitly define this.

Reply:
Updated